survey-api/src/main/java/org/kong/survey/service/SurveyService.java

```diff
@@ -25,64 +34,115 @@ public class SurveyService {
     @Autowired
     private QuestionRepository questionRepository;
 
+    @Autowired
+    private SurveyAnswerRepository surveyAnswerRepository;
+
     @Autowired
     private SurveyMapper surveyMapper;
 
     @Autowired
     private QuestionMapper questionMapper;
 
-    public List<SurveyFindAll.Response> findAll() {
-        List<SurveyEntity> surveyList = surveyRepository.findAll();
-        List<SurveyFindAll.Response> surveyFindAlls = surveyMapper.toSurveyFindAll(surveyList);
+    @Autowired
+    private SurveyAnswerMapper surveyAnswerMapper;
+
+    public Page<SurveyFindAll.Response> findAll(int page, int size) {
+        PageRequest pageRequest = PageRequest.of(page, size);
+        Page<SurveyEntity> surveyList = surveyRepository.findAll(pageRequest);
 
-        return surveyFindAlls;
+        List<SurveyFindAll.Response> surveyFindAlls = surveyMapper.toSurveyFindAll(surveyList.getContent());
+
+        Page<SurveyFindAll.Response> pageSurveyFindAlls = new PageImpl<>(surveyFindAlls, surveyList.getPageable(), surveyList.getTotalPages());
+
+        return pageSurveyFindAlls;
     }
 
     public Survey.Response findBySurveyId(Integer surveyId) {
-        SurveyEntity survey = surveyRepository.findBySurveyId(surveyId).orElseThrow(() -> new RuntimeException());
+        SurveyEntity survey = surveyRepository.findBySurveyId(surveyId).orElseThrow(() -> new RuntimeException("설문지를 찾을 수 없습니다."));
```

Comment:
```suggestion
        surveyRepository.findBySurveyId(surveyId).map(survey -> {
        }).then.....
```
메소드 체이닝을 사용해 보면 좋을 것 같아요.

Reply:
저는 Optional을 사용해서 객체 자체를 null을 확인 후, 뽑아내서 mapper에 넣어서 dto를 만드는 방식을 진행하는 데, map을 사용해서 어떤 이점을 얻을 수 있는 지 잘 모르겠습니다. 제가 개발을 구조를 잘 못 설정했을까요???
만약 SurveyEntity가 `@OneToMany`로 `List<QuestionEntity>`를 가져온다면 surveyRepository.findBySurveyId(surveyId).map(survey::getQuestions).orElseThrow(new RuntimeException());
이런 식으로 짜는 게 코드 가독성으로 좋을 것 같은데, 이 경우에는 메소드 체이닝으로 얻을 수 있게 뭔지 잘 모르겠습니다.